codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/StalledStreamProtectionConfigCustomization.kt

```diff
@@ -0,0 +1,169 @@
+
+    override fun configCustomizations(
+        codegenContext: ClientCodegenContext,
+        baseCustomizations: List<ConfigCustomization>,
+    ): List<ConfigCustomization> {
+        return baseCustomizations + StalledStreamProtectionConfigCustomization(codegenContext)
+    }
+
+    override fun operationCustomizations(
+        codegenContext: ClientCodegenContext,
+        operation: OperationShape,
+        baseCustomizations: List<OperationCustomization>,
+    ): List<OperationCustomization> {
+        return baseCustomizations + StalledStreamProtectionOperationCustomization(codegenContext)
+    }
+
+    override fun serviceRuntimePluginCustomizations(
+        codegenContext: ClientCodegenContext,
+        baseCustomizations: List<ServiceRuntimePluginCustomization>,
+    ): List<ServiceRuntimePluginCustomization> {
+        return baseCustomizations + StalledStreamProtectionRuntimePluginCustomization(codegenContext)
+    }
+
+    override fun extras(codegenContext: ClientCodegenContext, rustCrate: RustCrate) {
+        val rc = codegenContext.runtimeConfig
+
+        rustCrate.withModule(ClientRustModule.config) {
+            rustTemplate(
+                "pub use #{StalledStreamProtectionConfig};",
+                "StalledStreamProtectionConfig" to RuntimeType.smithyTypes(rc).resolve("stalled_stream_protection::StalledStreamProtectionConfig"),
+            )
+        }
+    }
+}
+
+/**
+ * Add a `stalled_stream_protection` field to Service config.
+ */
+class StalledStreamProtectionConfigCustomization(codegenContext: ClientCodegenContext) : NamedCustomization<ServiceConfig>() {
+    private val rc = codegenContext.runtimeConfig
+    private val codegenScope = arrayOf(
+        *preludeScope,
+        "StalledStreamProtectionConfig" to RuntimeType.smithyTypes(rc).resolve("stalled_stream_protection::StalledStreamProtectionConfig"),
+    )
+
+    override fun section(section: ServiceConfig): Writable {
+        return when (section) {
+            ServiceConfig.ConfigImpl -> writable {
+                rustTemplate(
+                    """
+                    /// Return a reference to the stalled stream protection configuration contained in this config, if any.
+                    pub fn stalled_stream_protection(&self) -> #{Option}<&#{StalledStreamProtectionConfig}> {
+                        self.config.load::<#{StalledStreamProtectionConfig}>()
+                    }
+                    """,
+                    *codegenScope,
+                )
+            }
+            ServiceConfig.BuilderImpl -> writable {
+                rustTemplate(
+                    """
+                    /// Set the [`StalledStreamProtectionConfig`](#{StalledStreamProtectionConfig})
+                    /// to configure protection for stalled streams.
+                    pub fn stalled_stream_protection(
+                        mut self,
+                        stalled_stream_protection_config: impl #{Into}<#{StalledStreamProtectionConfig}>
```

Comment:
is there anything that impls into here? If there's nothing actually a customer would put, probably better to accept it directly